src/components/add-journey/hooks.ts

```diff
@@ -0,0 +1,45 @@
+import axios from 'axios';
+import toast from 'react-hot-toast';
+import { TRANSPORT_API_URL } from '@/constants';
+import { useJourneySearchStore } from '@/hooks/useJourneySearchStore';
+import type { Journey, Station } from '@/types/opendata';
+import { useQuery } from '@tanstack/react-query';
+
+const getJourneys = ({
+  departureStation,
+  arrivalStation,
+  departureTime,
+}: {
+  readonly departureStation?: Station;
+  readonly arrivalStation?: Station;
+  readonly departureTime: string;
+}) => {
+  return axios.get<{ connections: Journey[] }>(
+    `${TRANSPORT_API_URL}/connections?from=${departureStation?.name}&to=${arrivalStation?.name}&date=${
+      departureTime.split('T')[0]
+    }&time=${departureTime.split('T')[1]}`
+  );
+};
+
+export const useGetJourneys = () => {
+  const setJourneys = useJourneySearchStore((state) => state.setJourneys);
+  const departureTime = useJourneySearchStore((state) => state.departureTime);
+  const departureStation = useJourneySearchStore((state) => state.departureStation);
+  const arrivalStation = useJourneySearchStore((state) => state.arrivalStation);
+
+  return useQuery(
```

Comment:
opting to use `useQuery` because it is already introduced as a dependency from `trpc`. Should simplify the api logic that was previously here